pyo3-macros-backend/src/pyenum.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2017-present PyO3 Project and Contributors
+
+use crate::common::impl_extractext;
+use crate::pyimpl::{PyClassImplBuilder, PyClassMethodsType};
+use proc_macro2::TokenStream;
+use quote::quote;
+
+pub fn build_py_enum(
+    enum_: &syn::ItemEnum,
+    methods_type: PyClassMethodsType,
+) -> syn::Result<TokenStream> {
+    let mut variants = Vec::new();
+
+    for variant in enum_.variants.iter() {
+        if !variant.fields.is_empty() {
+            return Err(syn::Error::new_spanned(
+                variant,
+                "#[pyenum] only supports unit enums",
+            ));
+        }
+        if let Some((_, syn::Expr::Lit(lit))) = &variant.discriminant {
+            variants.push((variant.ident.clone(), lit.clone()))
+        } else {
+            return Err(syn::Error::new_spanned(
+                variant,
+                "#[pyenum] requires explicit discriminant (MyVal = 4)",
+            ));
+        }
+    }
+
+    impl_enum(&enum_.ident, variants, methods_type)
+}
+
+fn impl_enum(
+    enum_: &syn::Ident,
+    variants: Vec<(syn::Ident, syn::ExprLit)>,
+    methods_type: PyClassMethodsType,
+) -> syn::Result<TokenStream> {
+    let enum_cls = impl_class(enum_, None, methods_type)?;
+    let variant_names: Vec<syn::Ident> = variants
+        .iter()
+        .map(|(ident, _)| variant_enumname(enum_, ident))
+        .collect::<syn::Result<Vec<_>>>()?;
+
+    let variant_cls = variants
+        .iter()
+        .map(|(ident, _)| impl_class(ident, Some(enum_), methods_type))
+        .collect::<syn::Result<Vec<_>>>()?;
```

Comment:
Does there need to be separate classes for the variants? I understood that was not the plan any more, maybe this is code to be killed off?

Reply:
Yes, I already killed that in my local repository.